tests/neo4j/datatypes/test_temporal_types.py

```diff
@@ -0,0 +1,1024 @@
+            types.CypherTime(0, 0, 0, 0, utc_offset_s=60 * 60 * -12),
+            # server enforced boundaries
+            types.CypherTime(23, 59, 59, 0, utc_offset_s=60 * 60 * 18),
+            types.CypherTime(0, 0, 0, 0, utc_offset_s=60 * 60 * -18),
+
+            types.CypherDateTime(1, 1, 1, 0, 0, 0, 0),
+            types.CypherDateTime(1970, 1, 1, 0, 0, 0, 0),
+            types.CypherDateTime(9999, 12, 31, 23, 59, 59, 999999999),
+            types.CypherDateTime(1970, 1, 1, 0, 0, 0, 0,
+                                 utc_offset_s=0),
+            types.CypherDateTime(1970, 1, 1, 0, 0, 0, 0,
+                                 utc_offset_s=60 * 60 * 18),
+            types.CypherDateTime(1970, 1, 1, 0, 0, 0, 0,
+                                 utc_offset_s=60 * 60 * -18),
+
+            # # FIXME: this breaks because the bolt protocol needs fixing
+            # # in the 80's the Swedes thought it was a good idea to introduce
+            # # daylight saving time. LOL, fools!
+            # # pre-shift 2:30 am UCT+2
+            # types.CypherDateTime(1980, 9, 28, 2, 30, 0, 0,
+            #                      utc_offset_s=60 * 60 * 2,
+            #                      timezone_id="Europe/Stockholm"),
+            # # one (non-political) hour later
+            # # post-shift 2:30 am again but UCT+1
+            # types.CypherDateTime(1980, 9, 28, 2, 30, 0, 0,
+            #                      utc_offset_s=60 * 60 * 1,
+            #                      timezone_id="Europe/Stockholm"),
+
+            types.CypherDuration(0, 0, 0, 0),
+            types.CypherDuration(0, 0, 0, -999999999),
+            types.CypherDuration(1, 2, 3, 4),
+            types.CypherDuration(-4, -3, -2, -1),
+            types.CypherDuration(0, 0, MAX_INT64, 999999999),
+            types.CypherDuration(0, MAX_INT64 // 86400, 0, 999999999),
+            types.CypherDuration(MAX_INT64 // 2629746, 0, 0, 999999999),
+            types.CypherDuration(0, 0, MIN_INT64, 0),
+            # Note: `int(a / b) != a // b`
+            #       `int(a / b)` rounds towards zero
+            #       `a // b` rounds down (towards negative infinity)
+            types.CypherDuration(0, int(MIN_INT64 / 86400), 0, 0),
+            types.CypherDuration(int(MIN_INT64 / 2629746), 0, 0, 0),
+        ]
+
+        self._create_driver_and_session()
+        for val in vals:
+            with self.subTest(val=val):
+                if get_driver_name() in ["python"]:
+                    if (isinstance(val, types.CypherDateTime)
+                            and val.utc_offset_s == 0):
+                        self.skipTest(
+                            "timezone library cannot tell the difference "
+                            "between named UTC and 0s offset timezone"
+                        )
+                self._verify_can_echo(val)
+
+    def test_should_echo_all_timezone_ids(self):
+        tz_ids = (
```

Comment:
not sure all of these will work across all languages
but I guess there is only one way to find out :)

Reply:
I'm not even sure if those are all the server supports :worried: It's just all Python supports - the 4 that the server reported it would not support... Maybe I should've checked elsewhere. I think the Java driver or SDN has tests that go over all timezone ids...